uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -93,11 +97,24 @@ define([
               startingDay: 1,
               showWeeks: false
             }
-          },
+          }
         }
       };
 
       /**
+       * Add a comment into comments array, also clears the comments textbox
+       */
+      this.addComment = function () {
+        this.request.comments.push({
+          contact_id: this.directiveOptions.contactId,
```

Comment:
No here the logged in users contact ID should be used. The leave request might be someone else's but the comment should be added for the currently logged in user.